### PR TITLE
MP-151 Creates an end point to return the category for a list of offe…

### PR DESCRIPTION
**Pull request by dave-virden**
Creates an end point to return the category for a list of offences

### CODE REVIEW

**Review by joelstobart-moj — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/ScheduleService.kt [hunk 1]
```diff
@@ -290,18 +313,32 @@ class ScheduleService(
...
+    val parts = schedulePartRepository.findByScheduleId(schedule15.id)
+    val part1Mappings = offenceScheduleMappingRepository.findBySchedulePartId(parts.first { p -> p.partNumber == 1 }.id)
+    val part2Mappings = offenceScheduleMappingRepository.findBySchedulePartId(parts.first { p -> p.partNumber == 2 }.id)
+    return Pair(part1Mappings, part2Mappings)
+  }
+
+  private fun getSexualOrViolentMappings(): Triple<List<OffenceScheduleMapping>, List<OffenceScheduleMapping>, List<OffenceScheduleMapping>> {
+    val scheduleThree = scheduleRepository.findOneByActAndCode("Sexual Offences Act 2003", "3")
+      ?: throw EntityNotFoundException("Schedule 3 not found")
+    val partThree = schedulePartRepository.findByScheduleId(scheduleThree.id)
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/ScheduleService.kt:336]: could replace lines 336 and 337 with a one line lookup 
`val scheduleThreeMapping = offenceScheduleMappingRepository.findBySchedulePartScheduleId(scheduleThree.id)`

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/ScheduleService.kt [hunk 2]
```diff
@@ -273,6 +277,25 @@ class ScheduleService(
     return getOffencePcscMarkers(offenceCodes)
   }
 
+  @Transactional(readOnly = true)
+  fun categoriseSexualOrViolentOffences(offenceCodes: List<String>): List<OffenceSexualOrViolent> {
+    log.info("Determining Sexual or Violent status for passed in offences")
+    return getSexualOrViolentIndicators(offenceCodes)
+  }
+  private fun getSexualOrViolentIndicators(offenceCodes: List<String>): List<OffenceSexualOrViolent> {
+    val (scheduleThreeMapping, part1Mappings, part2Mappings) = getSexualOrViolentMappings()
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/service/ScheduleService.kt:286]: can we rename scheduleThreeMapping to scheduleThreeMappings?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/model/OffenceSexualOrViolent.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  val offenceCode: String,
+  val schedulePart: OffenceSexualOrViolentIndicator,
+) {
+  companion object {
+    fun getSexualOrViolentIndicator(
+      inSchedule3: Boolean,
+      inSchedule15Part1: Boolean,
+      inSchedule15Part2: Boolean,
+    ): OffenceSexualOrViolentIndicator {
+      if (inSchedule15Part2 || inSchedule3) {
```
Comment (yasinmustafa) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/manageoffencesapi/model/OffenceSexualOrViolent.kt:19]: otlinb has a more concise version of doing this with less boilet plate; e.g.
```
return when {
        inSchedule15Part2 || inSchedule3 -> SEXUAL
        inSchedule15Part1 -> VIOLENT
        else -> NONE
    }
```

**Pull request merged**